internal/controllers/machineset/machineset_controller_status_test.go

```diff
@@ -0,0 +1,625 @@
+	machineSet := &clusterv1.MachineSet{}
+
+	tests := []struct {
+		name            string
+		machineSet      *clusterv1.MachineSet
+		machines        []*clusterv1.Machine
+		expectCondition metav1.Condition
+	}{
+		{
+			name:       "getAndAdoptMachines failed",
+			machineSet: machineSet,
+			machines:   nil,
+			expectCondition: metav1.Condition{
+				Type:    clusterv1.MachineSetMachinesUpToDateV1Beta2Condition,
+				Status:  metav1.ConditionUnknown,
+				Reason:  clusterv1.MachineSetMachinesUpToDateInternalErrorV1Beta2Reason,
+				Message: "Please check controller logs for errors",
+			},
+		},
+		{
+			name:       "One machine up-to-date",
+			machineSet: machineSet,
+			machines: []*clusterv1.Machine{
+				createMachine("up-to-date-1", metav1.Condition{
+					Type:   clusterv1.MachineUpToDateV1Beta2Condition,
+					Status: metav1.ConditionTrue,
+					Reason: "some-reason-1",
+				}),
+			},
+			expectCondition: metav1.Condition{
+				Type:    clusterv1.MachineSetMachinesUpToDateV1Beta2Condition,
+				Status:  metav1.ConditionTrue,
+				Reason:  "some-reason-1",
+				Message: "",
+			},
+		},
+		{
+			name:       "One machine unknown",
+			machineSet: machineSet,
+			machines: []*clusterv1.Machine{
+				createMachine("unknown-1", metav1.Condition{
+					Type:    clusterv1.MachineUpToDateV1Beta2Condition,
+					Status:  metav1.ConditionUnknown,
+					Reason:  "some-unknown-reason-1",
+					Message: "some unknown message",
+				}),
+			},
+			expectCondition: metav1.Condition{
+				Type:    clusterv1.MachineSetMachinesUpToDateV1Beta2Condition,
+				Status:  metav1.ConditionUnknown,
+				Reason:  "some-unknown-reason-1",
+				Message: "some unknown message from Machine unknown-1",
+			},
+		},
+		{
+			name:       "One machine not up-to-date",
+			machineSet: machineSet,
+			machines: []*clusterv1.Machine{
+				createMachine("not-up-to-date-machine-1", metav1.Condition{
+					Type:    clusterv1.MachineUpToDateV1Beta2Condition,
+					Status:  metav1.ConditionFalse,
+					Reason:  "some-not-up-to-date-reason",
+					Message: "some not up-to-date message",
+				}),
+			},
+			expectCondition: metav1.Condition{
+				Type:    clusterv1.MachineSetMachinesUpToDateV1Beta2Condition,
+				Status:  metav1.ConditionFalse,
+				Reason:  "some-not-up-to-date-reason",
+				Message: "some not up-to-date message from Machine not-up-to-date-machine-1",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			setMachinesUpToDateCondition(ctx, tt.machineSet, tt.machines, tt.machines != nil)
+
+			condition := v1beta2conditions.Get(tt.machineSet, clusterv1.MachineSetMachinesUpToDateV1Beta2Condition)
+			g.Expect(condition).ToNot(BeNil())
+			g.Expect(*condition).To(v1beta2conditions.MatchCondition(tt.expectCondition, v1beta2conditions.IgnoreLastTransitionTime(true)))
+		})
+	}
+}
+
+func createMachine(name string, conditions ...metav1.Condition) *clusterv1.Machine {
```

Comment:
Maybe it makes sense to rename these funcs:
* createMachine => machine()
* testMachine => staleDeletingMachine()

(and then only use the latter in cases where we set staleDeleting to true at the moment + get rid of the parameter)

the "create" in "createMachine" just seems a bit strange in comparison to "testMachine" (both are just constructing a Machine object, neither are creating it anywhere)

Reply:
Using newMachine and newStaleDeletingMachine as we have in oder tests.

Cannot use machine due to it being a name of a imported package.

Reply:
Makes sense!

Reply:
Using newMachine and newStaleDeletingMachine as we have in oder tests.

Cannot use machine due to it being a name of a imported package.